### PR TITLE
ELECTRON-1170 (Remove electron-remote dependency)

### PR DESCRIPTION
**Pull request by KiranNiranjan**
## Description
Fix few javascript errors
[ELECTRON-1170](https://perzoinc.atlassian.net/browse/ELECTRON-1170)
[ELECTRON-1185](https://perzoinc.atlassian.net/browse/ELECTRON-1185)

## Solution Approach

- Remove deprecated `electron-remote` module
- Update target version
- Fix `menu.popup` options
- Remove ContextMenuListener logic

### CODE REVIEW

**Review by keerthi16 — APPROVED** [review]

File: src/dictionary-sync.js [hunk 1]
```diff
@@ -71,45 +65,23 @@ module.exports = class DictionarySync {
...
-
-        if (ret.length < 8*1024) {
-          throw new Error("File exists but is most likely bogus");
-        }
+    if (cacheOnly) return target;
 
-        return ret;
+    try {
+      if (!fs.existsSync(target)) {
+        throw new Error(`${lang} is currently not supported`);
```
Comment (VishwasShashidhar) [inline src/dictionary-sync.js:72]: will this quit the app?

File: src/dictionary-sync.js [hunk 2]
```diff
@@ -71,45 +65,23 @@ module.exports = class DictionarySync {
...
+      let ret = await fs.readFile(target, {});
 
-    if (fileExists) {
-      try {
-        await fs.unlink(target);
-      } catch (e) {
-        d("Can't clear out file, bailing");
-        throw e;
+      if (ret.length < 8*1024) {
+        throw new Error("File exists but is most likely bogus");
```
Comment (VishwasShashidhar) [inline src/dictionary-sync.js:78]: in such a case, can we delete the file instead?

**Review by VishwasShashidhar — CHANGES_REQUESTED** [review]

File: src/dictionary-sync.js (hunk 2 above)
Comment (KiranNiranjan) [reply src/dictionary-sync.js:78]: This won't happen because we are manually copying those files now

File: src/dictionary-sync.js (hunk 1 above)
Comment (KiranNiranjan) [reply src/dictionary-sync.js:72]: No, this just to debug

**Review by VishwasShashidhar — APPROVED** [review]

**Pull request merged**